src/dash-adapter.js

```diff
@@ -934,7 +936,12 @@ export default class DashAdapter extends BaseMediaSourceAdapter {
    * @public
    */
   selectTextTrack(textTrack: TextTrack): void {
-    if (this._shaka && textTrack instanceof TextTrack && !textTrack.active && (textTrack.kind === 'subtitles' || textTrack.kind === 'captions')) {
+    if (
+      this._shaka &&
+      textTrack instanceof TextTrack &&
+      !textTrack.active &&
+      (textTrack.kind === 'subtitles' || textTrack.kind === 'captions' || this._config.useShakaTextTrackDisplay)
```

Comment:
why do we need to add this to the OR statement?

Reply:
I saw the ticket description now - it looks more like a hole that we need to fix rather than adding this flag here.

Reply:
done